flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannelTest.java

```diff
@@ -431,8 +426,35 @@ public void testGetNextAfterPartitionReleased() throws Exception {
 		assertFalse(channel.getNextBuffer().isPresent());
 	}
 
+	/**
+	 * Verifies that buffer is not compressed when getting from a {@link LocalInputChannel}.
+	 */
+	@Test
+	public void testGetBufferFromLocalChannelWhenCompressionEnabled() throws Exception {
+		ResultSubpartitionView subpartitionView = createResultSubpartitionView(true);
+		TestingResultPartitionManager partitionManager = new TestingResultPartitionManager(subpartitionView);
+		LocalInputChannel channel = createLocalInputChannel(new SingleInputGateBuilder().build(), partitionManager);
+
+		// request partition and get next buffer
+		channel.requestSubpartition(0);
+		Optional<InputChannel.BufferAndAvailability> bufferAndAvailability = channel.getNextBuffer();
+		assertTrue(bufferAndAvailability.isPresent());
+		assertFalse(bufferAndAvailability.get().buffer().isCompressed());
+	}
+
 	// ---------------------------------------------------------------------------------------------
 
+	private static ResultSubpartitionView createResultSubpartitionView(boolean addBuffer) throws IOException {
+		int bufferSize = 4096;
+		ResultPartition parent = PartitionTestUtils.createPartition(
+			ResultPartitionType.PIPELINED, NoOpFileChannelManager.INSTANCE, true, "LZ4", bufferSize);
+		ResultSubpartition subpartition = new PipelinedSubpartition(0, parent);
+		if (addBuffer) {
+			subpartition.add(BufferBuilderTestUtils.createFilledFinishedBufferConsumer(bufferSize));
```

Comment:
Do not need to create `subpartition` above and use `parent.getAllPartitions()[0]` instead, then we do not change the package public for `PipelinedSubpartition` class.